x-pack/plugins/apm/public/components/app/settings/agent_explorer/agent_instances/agent_instances_details/index.tsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLoadingContent } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ValuesType } from 'utility-types';
+import { AgentExplorerFieldName } from '../../../../../../../common/agent_explorer';
+import { getServiceNodeName } from '../../../../../../../common/service_nodes';
+import { APIReturnType } from '../../../../../../services/rest/create_call_apm_api';
+import { unit } from '../../../../../../utils/style';
+import { EnvironmentBadge } from '../../../../../shared/environment_badge';
+import { ItemsBadge } from '../../../../../shared/item_badge';
+import { ServiceNodeMetricOverviewLink } from '../../../../../shared/links/apm/service_node_metric_overview_link';
+import {
+  ITableColumn,
+  ManagedTable,
+} from '../../../../../shared/managed_table';
+import { TimestampTooltip } from '../../../../../shared/timestamp_tooltip';
+import { TruncateWithTooltip } from '../../../../../shared/truncate_with_tooltip';
+
+type AgentExplorerInstance = ValuesType<
+  APIReturnType<'GET /internal/apm/services/{serviceName}/agent_instances'>['items']
+>;
+
+enum AgentExplorerInstanceFieldName {
+  InstanceName = 'serviceNode',
+  Environments = 'environments',
+  AgentName = 'agentName',
+  AgentVersion = 'agentVersion',
+  LastReport = 'lastReport',
+}
+
+export function getInstanceColumns(
+  serviceName: string
+): Array<ITableColumn<AgentExplorerInstance>> {
+  return [
+    {
+      field: AgentExplorerInstanceFieldName.InstanceName,
+      name: i18n.translate(
+        'xpack.apm.agentExplorerInstanceTable.InstanceColumnLabel',
+        {
+          defaultMessage: 'Instance',
+        }
+      ),
+      sortable: true,
+      render: (_, { serviceNode }) => {
+        const { displayedName, tooltip } = !serviceNode
+          ? {
+              displayedName: getServiceNodeName(serviceNode),
+              tooltip: i18n.translate(
+                'xpack.apm.agentExplorerInstanceTable.explainServiceNodeNameMissing',
+                {
+                  defaultMessage:
+                    'This agent instance is not tied to a service node.',
```

Comment:
I'm not sure if this is helpful to the user. It's about them setting `serviceNodeName`: https://www.elastic.co/guide/en/apm/agent/nodejs/master/configuration.html#service-node-name. We should probably link to the docs.

Reply:
Thank you for your feedback. In order to have an interactive popup and place the link I had to implement a Popover component
<img width="849" alt="image" src="https://user-images.githubusercontent.com/1313018/201648784-6c9b8f1c-9463-42f2-a6d2-e32d26a441e8.png">.

When the serviceNodeName is informed the user will still see the normal popoup
<img width="774" alt="image" src="https://user-images.githubusercontent.com/1313018/201649171-4a917b6b-9709-4bd7-bfbc-b65eb46627b8.png">

What do you think?

Reply:
That's great I think 👍